src/mbgl/gl/context.cpp

```diff
@@ -480,6 +481,14 @@ bool Context::setupDraw(const PaintParameters& parameters, const gfx::Drawable&
     setColorMode(parameters.colorModeForRenderPass());
     setCullFaceMode(gfx::CullFaceMode::disabled());
 
+    int32_t nextTextureUnit = 0;
+    for (auto& attachment : drawable.getTextures()) {
+        std::static_pointer_cast<gl::Texture2D>(attachment.texture)->bind(attachment.location, nextTextureUnit++);
```

Comment:
Do we want to bind textures here or in `DrawableGL::draw`? It's actually doing both at the moment because I forgot to pick one.

Reply:
I prefer drawables for this kind of thing, but I defer to the rest of you if you want to keep the Context alive.